File: assign-cookies/assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        if (s.empty()) return 0;
        
        int res = 0;
        int idx = s.size() - 1;
        
        sort(s.begin(), s.end());
        sort(g.begin(), g.end());
        
        for (int i = g.size() - 1; i >=0; i--){
            if (idx >= 0 && s[idx] >= g[i]){
                res++; idx--;
            }
        }
        return res;
    }
};

File: assign-cookies/assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        if not s: return 0
        
        count = 0
        s.sort()
        g.sort()
        
        while s and g:
            if s[-1] >= g[-1]: 
                count += 1
                s.pop(), g.pop()
            else:
                g.pop()

        return count